### PR TITLE
Follow multiple redirects

### PR DESCRIPTION
**Pull request by bekir-ozturk**
Some aka.ms links are redirecting to other aka.ms links. This change makes sure we follow them until we reach the final URL.

### CODE REVIEW

**Comment by bekir-ozturk** [comment]
CI run in aspnetcore repo for the changes in this PR are here: https://github.com/dotnet/aspnetcore/pull/32845
So far: 21 passed, 3 in progress.

File: src/dotnet-install.ps1 [hunk 1]
```diff
@@ -858,39 +858,53 @@ function Get-AkaMSDownloadLink([string]$Channel, [string]$Quality, [bool]$Intern
...
+                $akaMsDownloadLink = $Response.Headers.GetValues("Location")[0]
+
+                if ([string]::IsNullOrEmpty($akaMsDownloadLink)) {
+                    Say-Verbose "The aka.ms link '$akaMsLink' is not valid: failed to get redirect location."
+                    return $null
+                }
 
-            if ([string]::IsNullOrEmpty($akaMsDownloadLink)) {
+                Say-Verbose "The redirect location retrieved: '$akaMsDownloadLink'."
+                if ($akaMsDownloadLink.StartsWith('https://aka.ms/'))
```
Comment (vlada-shubina) [inline src/dotnet-install.ps1:887]: as discussed, I don't like this limitation of comparing link - this makes solution more fragile, moreover no same limitation in bash.

let's try to reverse to redirect to the last but one and take link.

File: src/dotnet-install.ps1 [hunk 2]
```diff
@@ -858,39 +858,54 @@ function Get-AkaMSDownloadLink([string]$Channel, [string]$Quality, [bool]$Intern
...
+            # We ended up with a valid URL. This is where the resource should be.
+            return $akaMsDownloadLink
         }
+
+        Say-Verbose "The link '$akaMsLink' is not valid: failed to trace redirections to a valid resource. Last received HTTP code is '$Response.StatusCode'."
+        return $null
     }
-    Say-Verbose "The aka.ms link '$akaMsLink' is not valid: failed to get redirect location. The resourse is not available."
+
+    Say-Warning "Aka.ms links have redirected more than the maximum allowed redirections. This may be caused by a cyclic redirection of aka.ms links."
```
Comment (vlada-shubina) [inline src/dotnet-install.ps1:906]: `Say-Verbose` probably fits better, as all errors above are reported to `Verbose`

**Review by vlada-shubina — APPROVED** [review]

File: src/dotnet-install.ps1 (hunk 2 above)
Comment (bekir-ozturk) [reply src/dotnet-install.ps1]: Done

File: src/dotnet-install.ps1 [hunk 3]
```diff
@@ -858,39 +858,54 @@ function Get-AkaMSDownloadLink([string]$Channel, [string]$Quality, [bool]$Intern
...
                 return $null
             }
-
-            Say-Verbose "The redirect location retrieved: '$akaMsDownloadLink'."
-            return $akaMsDownloadLink
         }
-        catch {
-            Say-Verbose "The aka.ms link '$akaMsLink' is not valid: failed to get redirect location."
-            return $null
+        elseif ($Response.StatusCode -eq 200)
```
Comment (AR-May) [inline src/dotnet-install.ps1:896]: Are you sure that 200 as a response code is the only success code you would receive? Maybe better to check the whole diapason of success codes?

File: src/dotnet-install.ps1 [hunk 4]
```diff
@@ -858,39 +858,54 @@ function Get-AkaMSDownloadLink([string]$Channel, [string]$Quality, [bool]$Intern
...
+        {
+            # We ended up with a valid URL. This is where the resource should be.
+            return $akaMsDownloadLink
         }
+
+        Say-Verbose "The link '$akaMsLink' is not valid: failed to trace redirections to a valid resource. Last received HTTP code is '$Response.StatusCode'."
```
Comment (AR-May) [inline src/dotnet-install.ps1:902]: The internal builds would end up here and would not be resolved as far as I can see.

**Review by AR-May — CHANGES_REQUESTED** [review]

File: src/dotnet-install.ps1 (hunk 3 above)
Comment (bekir-ozturk) [reply src/dotnet-install.ps1]: We are now accepting any code as long as it is not a 301. All 2xx codes are covered.

File: src/dotnet-install.ps1 (hunk 4 above)
Comment (bekir-ozturk) [reply src/dotnet-install.ps1]: Should be fixed in latest commit.

**Review by AR-May — APPROVED** [review]
LGTM

**Pull request merged**